### PR TITLE
Fixes for APPM-1365

### PR DESCRIPTION
**Pull request by lakshani**

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 1]
```diff
@@ -259,6 +233,42 @@ public void deleteBusinessOwner(String businessOwnerId) throws AppManagementExce
...
 
     /**
+     * Check the business owner usage of web app from the registry.
+     * @param businessOwnerId
+     * @param registry
+     * @param tenantDomain
+     * @return
+     * @throws AppManagementException
+     */
+    public boolean isAssociatedWithApps(String businessOwnerId, Registry registry, String tenantDomain) throws
```
Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java]: Rename this method name to isAssociatedWithApps -> isBusinessOwnerAssociatedWithApps()

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 2]
```diff
@@ -233,23 +207,23 @@ public void deleteBusinessOwner(String businessOwnerId) throws AppManagementExce
...
             statementToDeleteRecordTwo.setString(1, businessOwnerId);
             statementToDeleteRecordTwo.executeUpdate();
 
-            String queryToDeleteRecord = "DELETE FROM BUSINESS_OWNER WHERE OWNER_ID = ?";
+            String queryToDeleteRecord = "DELETE FROM APM_BUSINESS_OWNER WHERE OWNER_ID = ?";
 
             statementToDeleteRecord = connection.prepareStatement(queryToDeleteRecord);
             statementToDeleteRecord.setString(1, businessOwnerId);
-            statementToDeleteRecord.executeUpdate();
+            int i = statementToDeleteRecord.executeUpdate();
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:220]: unnecessary assignment.

Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:220]: Fixed in https://github.com/wso2/carbon-appmgt/commit/9b1e766e9063423faaf1b86f353a0db0d54deb12

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js [hunk 3]
```diff
@@ -275,15 +275,26 @@ $(document).on("click", ".owner-edit-button", function () {
...
     $.ajax({
                url: context + '/apis/businessowners/' + ownerId,
                type: 'DELETE',
                contentType: 'application/json',
                dataType: 'json',
                success: function (response) {
-                   updateOwners();
-                   Showalert("Owner Deleted Successfully ", "alert-success", "statusSuccess");
-                   location.reload();
+                   if(response) {
```
Comment (Kishanthan) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js:285]: Not formatted

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js [hunk 4]
```diff
@@ -275,15 +275,26 @@ $(document).on("click", ".owner-edit-button", function () {
...
+                       $(".alert").alert();
+                       var isConfirmed = confirm("Are you sure you want to delete the business owner " + ownerName + "?");
+                       if (isConfirmed) {
+                           updateOwners();
+                           Showalert("Business Owner : " + ownerName + " Deleted Successfully ", "alert-success", "statusSuccess");
```
Comment (Kishanthan) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js:290]: This alert msg is not consistent. Aa a best practice, we don't start intermediate words with "Caps" with log or alert msges as they are shown to the user.

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java [hunk 5]
```diff
@@ -128,7 +137,7 @@ public void updateBusinessOwner(BusinessOwner businessOwner) throws AppManagemen
      */
     @Override
     public List<BusinessOwner> getBusinessOwners() throws AppManagementException {
-        return appMDAO.getBusinessOwners();
+        return appMDAO.getBusinessOwners(tenantId);
```
Comment (Kishanthan) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:140]: where do we get the tenant id here? And how do we get the tenant id?

Comment (lakshani) [reply components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:140]: It is defined in a super class.

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js (hunk 3 above)
Comment (lakshani) [reply features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js:285]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/665ba5bee36f67c4e163ac0422175545ad3a03c5

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js (hunk 4 above)
Comment (lakshani) [reply features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js:290]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/665ba5bee36f67c4e163ac0422175545ad3a03c5

File: components/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java [hunk 6]
```diff
@@ -40,7 +40,7 @@
      * @param businessOwnerId ID of the owner.
      * @throws AppManagementException
      */
-    public void deleteBusinessOwner(String businessOwnerId) throws AppManagementException;
+    public boolean deleteBusinessOwner(String businessOwnerId) throws AppManagementException;
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java:43]: Java doc comments issues and formatting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java [hunk 7]
```diff
@@ -104,11 +104,20 @@ public APIProviderImpl(String username) throws AppManagementException {
     /**
      * Delete business owner.
      * @param businessOwnerId ID of the owner.
+     * @return
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:107]: Java doc comments issues and formatting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files

File: components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 8]
```diff
@@ -259,6 +233,42 @@ public void deleteBusinessOwner(String businessOwnerId) throws AppManagementExce
     }
 
     /**
+     * Check the business owner usage of web app from the registry.
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:236]: Java doc comments issues and formatting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files

**Pull request merged**